client/src/featureform/status_display.py

```diff
@@ -0,0 +1,156 @@
+            name=resource.name,
+            variant=variant,
+            status=resource.status,
+            error=resource.error,
+        )
+
+
+class StatusDisplayer:
+    resource_types_to_check = {
+        Feature,
+        OnDemandFeature,
+        TrainingSet,
+        Label,
+        Source,
+        Provider,
+    }
+
+    def __init__(self, stub: ApiStub, resources: List[Resource]):
+        filtered_resources = filter(
+            lambda r: type(r) in self.resource_types_to_check, resources
+        )
+        self.stub = stub
+
+        # A more intuitive way to is to store OrderedDict[Resource, DisplayStatus] but you can't hash Resource easily
+        self.resource_to_status_list: List[Tuple[Resource, DisplayStatus]] = []
+
+        for r in filtered_resources:
+            self.resource_to_status_list.append((r, DisplayStatus.from_resource(r)))
+
+    def update_display_statuses(self):
+        for resource, display_status in self.resource_to_status_list:
+            if resource.name == "local-mode":
+                continue
+            if not display_status.is_finished():
+                r = resource.get(self.stub)
+                display_status.status = r.status
+                display_status.error = r.error
+
+    def all_statuses_finished(self) -> bool:
+        return all(status.is_finished() for _, status in self.resource_to_status_list)
+
+    def display(self):
+        if not self.resource_to_status_list:
+            return
+
+        print()
+        console = Console()
+        with Live(console=console, auto_refresh=True, screen=False) as live:
+            i = 0
+            while True:
+                self.update_display_statuses()
+                finished_running = self.all_statuses_finished()
+
+                dots = "." * (1 + i % MAX_NUM_RUNNING_DOTS)
+
+                title = (
+                    f"[green]COMPLETED[/]"
+                    if finished_running
+                    else f"[yellow]RUNNING{dots}[/]"
+                )
+                table = Table(
+                    title=title,
+                    title_justify="left",
+                    show_header=True,
+                    header_style="bold",
+                    box=None,
+                )
+                table.add_column("Resource Type", width=25)
+                table.add_column("Name (Variant)", width=50, no_wrap=True)
+                table.add_column("Status", width=10)
+                table.add_column("Error", style="red")
+
+                for _, status in self.resource_to_status_list:
+                    error = f" {status.error}" if status.error else ""
+                    resource_type = status.resource_type.__name__
+                    name = status.name
+                    status_text = (
+                        status.status
+                        if status.resource_type is not Provider
+                        else "CREATED"
+                    )
+
+                    status_to_color = {
```

Comment:
since this doesn't change, does it make sense to move it outside of the loop? we can make it constants too

Reply:
yeahh good call